app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -359,12 +360,26 @@ class FormController(
       choice.bindFromRequest
         .fold(
           _ =>
-            BadRequest(continue_form_page(cache.formTemplate, maybeAccessCode, lang, frontendAppConfig)).pure[Future], {
+            BadRequest(
+              continue_form_page(
+                cache.formTemplate,
+                choice.bindFromRequest().withError("decision", "error.required"),
+                maybeAccessCode,
+                lang,
+                frontendAppConfig)).pure[Future], {
             case "continue" =>
               val dataRaw = FormDataHelpers.formDataMap(cache.form.formData)
               redirectWithRecalculation(cache, dataRaw, maybeAccessCode, lang)
             case "delete" =>
-              Ok(confirm_delete(cache.formTemplate, maybeAccessCode, lang, frontendAppConfig)).pure[Future]
+              gformConnector
```

Comment:
Can't you use uk.gov.hmrc.gform.gform.FormController#delete, rather than repeating yourself?

Reply:
Because the delete method returned a different type (Action[AnyContent]) I wasn't able to use it directly but after consulting with @josef-vlach we have avoided the DRY principle by using a standalone method used by both functions that returns a Future[Result] - also we have removed the unused template (confirm_delete) and route to avoid future confusion